### PR TITLE
feat(button): setup button component

### PR DESCRIPTION
**Pull request by andresin87**
### Types of changes

- [ ] 🛠️ Tool
- [ ] 🪲 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🧾 Documentation
- [ ] 📷 Demo
- [ ] 🧪 Test
- [ ] 🧠 Refactor
- [ ] 💄 Styles

### CODE REVIEW

File: packages/components/button/src/Button.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+<StoryHeading label="Import" />
+
+```js
+import { Button } from "@spark-ui/button"
+```
+
+<StoryHeading label="Props" />
+
+<ArgsTable of={Button} />
```
Comment (andresz1) [inline packages/components/button/src/Button.stories.mdx:28]: Normally props are placed in the bottom of the documentation

File: packages/components/button/src/Button.test.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+describe('Button', () => {
+  it('should render', () => {
+    // Given
+
+    // When
+    render(<Button>Hello World!</Button>)
+
+    // Then
+    expect(screen.getByText('Hello World!')).toBeInTheDocument()
```
Comment (andresz1) [inline packages/components/button/src/Button.test.tsx:15]: Use `getByRole('button', {name: ''})` instead 🙏🏻

File: packages/components/button/src/Button.test.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  })
+
+  it('should trigger click event', async () => {
+    // Given
+    const user = userEvent.setup()
+    const clickEvent = vi.fn()
+
+    // When
+    render(<Button onClick={clickEvent}>Hello World!</Button>)
+    await user.click(screen.getByText('Hello World!'))
```
Comment (andresz1) [inline packages/components/button/src/Button.test.tsx:25]: Same as above

File: packages/components/button/src/Button.test.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    const user = userEvent.setup()
+    const clickEvent = vi.fn()
+
+    // When
+    render(
+      <Button onClick={clickEvent} disabled>
+        Hello World!
+      </Button>
+    )
+    await user.click(screen.getByText('Hello World!'))
```
Comment (andresz1) [inline packages/components/button/src/Button.test.tsx:42]: Same as above

Comment (andresz1) [inline packages/components/button/src/Button.test.tsx:42]: I would also check for `disabled` attribute

File: packages/components/button/src/Button.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import React, { PropsWithChildren } from 'react'
+
+import { buttonStyles, type ButtonStylesProps } from './Button.styles'
+
+/** Review: Prop VS ButtonProps */
+interface Props
```
Comment (andresz1) [inline packages/components/button/src/Button.tsx:6]: IMO you should export button `Props`

File: packages/components/button/src/variants/tinted.ts [hunk 6]
```diff
@@ -0,0 +1,88 @@
+const tw = <T>(a: T): T => a
```
Comment (andresz1) [inline packages/components/button/src/variants/tinted.ts:1]: This function definition is repeated several times. Can we do something regarding that ? I think that is not needed

**Review by andresz1 — CHANGES_REQUESTED** [review]

File: packages/components/button/package.json [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  "name": "@spark-ui/button",
+  "version": "1.0.0",
+  "description": "Button component is used to trigger an action or event, such as submitting a form, opening a Dialog, canceling an action, or performing a delete operation.",
+  "publishConfig": {
+    "access": "public"
+  },
+  "main": "./dist/index.js",
+  "module": "./dist/index.mjs",
+  "types": "./dist/index.d.ts",
+  "scripts": {
```
[the remark this answers is not in the document]
Comment (andresz1) [reply packages/components/button/package.json:11]: Also I think that react should be a [peerDependency](https://github.com/chakra-ui/chakra-ui/blob/main/packages/components/accordion/package.json#L59)

File: packages/components/button/src/Button.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+  'success',
+  'alert',
+  'danger',
+  'neutral',
+  'surface',
+]
+const designs: ButtonProps['design'][] = ['filled', 'outlined', 'tinted', 'ghost', 'contrast']
+const shapes: ButtonProps['shape'][] = ['rounded', 'square', 'pill']
+
+export const Sizes: StoryFn = () => (
```
Comment (andresz1) [inline packages/components/button/src/Button.stories.tsx:23]: I think that you need to handle `args` here or the code documentation won't work. See https://github.com/adevinta/spark/blob/main/packages/components/radio/src/Radio.stories.tsx#L6

File: packages/components/button/src/Button.tsx [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+      intent = 'primary',
+      shape = 'rounded',
+      size = 'md',
+      ...rest
+    },
+    ref
+  ) => {
+    return (
+      <button
+        role="button"
```
Comment (andresz1) [inline packages/components/button/src/Button.tsx:24]: Is role needed ?

File: packages/components/button/src/Button.tsx [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+      disabled = false,
+      intent = 'primary',
+      shape = 'rounded',
+      size = 'md',
+      ...rest
+    },
+    ref
+  ) => {
+    return (
+      <button
```
Comment (andresz1) [inline packages/components/button/src/Button.tsx:23]: I think that [slot](https://www.radix-ui.com/docs/primitives/utilities/slot) component should be used here. Normally you want anchors look like buttons. Here you have an [example](https://www.radix-ui.com/docs/primitives/utilities/slot#usage)

File: packages/components/button/src/Button.tsx [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+
+export const Button = React.forwardRef<HTMLButtonElement, PropsWithChildren<Props>>(
+  (
+    {
+      design = 'filled',
+      disabled = false,
+      intent = 'primary',
+      shape = 'rounded',
+      size = 'md',
+      ...rest
```
Comment (andresz1) [inline packages/components/button/src/Button.tsx:18]: Should we use `rest` or `others` ?

File: packages/components/button/src/Button.test.tsx (hunk 2 above)
Comment (andresin87) [reply packages/components/button/src/Button.test.tsx:15]: OMG. Maybe we should change the generator, cuz it's the 2nd time we got the same error

File: packages/components/button/src/Button.tsx (hunk 9 above)
Comment (andresin87) [reply packages/components/button/src/Button.tsx:24]: nope

File: packages/components/button/src/Button.tsx (hunk 10 above)
Comment (andresin87) [reply packages/components/button/src/Button.tsx:23]: yes. Its a primitive we might code soon

File: packages/components/button/src/Button.tsx (hunk 11 above)
Comment (andresin87) [reply packages/components/button/src/Button.tsx:18]: i prefer `otherProps`

**Review by andresz1 — APPROVED** [review]

**Pull request merged**